app/controllers/users/couches_controller.rb

```diff
@@ -31,4 +32,8 @@ def couch_params
                                     :zipcode,
                                     :host)
     end
+
+    def verify_user
+      render file: "public/404", status: 404 unless current_user && current_user.id == params["user_id"].to_i
```

Comment:
@tmikeschu What would you think about breaking `current_user && current_user.id == params["user_id"].to_i` out into a method?  Maybe it could read something like `404 unless the_current_user_is_viewing_their_own_information`.  (If you make the method names ugly enough you will be motivated to find better ones.)